### DISCUSSION TITLE
Logo Image place/URL and Submenu

### DISCUSSION THREAD

**Discussion by horieva** [gethinode/hinode#928] [Help]
Hello there. I have two questions. Sorry for dumb question because I'm novice in Hinode/Hugo. 

1. I've created a new logo and placed it in /assets/img/new-logo.svg. However, when I specify the URL "/img/new-logo.svg" in /config/_default/params.toml, I get a 404 error. Why is that?

```
[main]
    featurePhoto = "/img/sunrise.jpg" # source: https://unsplash.com/photos/ZX6BPboJrYk
    logo = "/img/new-logo.svg"
```

2. On our website, I'd like to implement a dropdown menu. How can I achieve this? And is it possible to make the items in this menu not separate pages but rather links to anchors #?

Menu structure:
```
Page (/page/)
- Item 1 (/page/#item1)
- Item 2 (/page/#item2)
etc
```

3. When I change as described in documentation (we want two languages EN and NL) `DefaultContentLanguageInSubdir = true` to `no` in config/_default/config.toml my site starting loop from / to /en. How I can fix that? I won't /en for the primary language. 

Thank you for the great theme!

**Comment by markdumay** [comment]
Glad you like the theme! No worries about your questions, we're here to help. However, it might be helpful to review the available documentation on https://gethinode.com first. 

The vector logo needs to be in the `static` folder. See https://gethinode.com/docs/configuration/navigation/#basic-configuration for more details.

You can find an example of a nested menu in the docs: https://gethinode.com/docs/configuration/navigation/#menus.

The [language docs ](https://gethinode.com/docs/configuration/languages/) explain how to configure the site's languages, and how to discard an existing language entirely.

**Comment by horieva** [comment]
Thank you, now logo and languages directory displayed ok!

Still have problem with nested menus. My code is: 

```
[[main]]
  name = "Services"
  url = "/services/"
  weight = 20

  [[main]]
  parent = "Services"
  name = "Renovations"
  url = "/services/#renovations"
  weight = 21

  [[main]]
  parent = "Services"
  name = "Extensions"
  url = "/services/#extensions"
  weight = 22
```

No dropdown. What I'm doing wrong?

**Reply by markdumay** [reply]
You might want to check the indentation, each ‘[[main]]’ should start at the beginning of a new line.

**Reply by horieva** [reply]
Don't work. Can I show my repo to you? 
https://github.com/horieva/mpe-bouwtechniek
Thank you!

**Reply by markdumay** [reply]
Sure - I’ll have a look at it

**Reply by horieva** [reply]
Netlify deploy: https://eliza.horieva.com

**Reply by markdumay** [reply]
You're using an obsolete version of Hinode which is no longer maintained (`@markdumay/hugo-theme-hinode`). Please check the [getting started](https://gethinode.com/docs/getting-started/introduction/) section of the docs. There's also a guide on how to [create your first site](https://gethinode.com/guides/editing/).

The menu configuration works in the current version of Hinode:

```toml
[[main]]
  name = "Services"
  url = "/services/"
  weight = 20

[[main]]
  parent = "Services"
  name = "Renovations"
  url = "/services/#renovations"
  weight = 21

[[main]]
  parent = "Services"
  name = "Extensions"
  url = "/services/#extensions"
  weight = 22
```

![localhost_1313_(MacBook Pro 14_ (scaled)) (2)](https://github.com/gethinode/hinode/assets/61946753/3ff3c59a-5433-497e-a9e6-9ceed63cb16c)

**Reply by horieva** [reply]
Oh that explains a lot! I’m sorry I’m confused in the versions of Hinode. Please tell me how to transfer what I have already done to the new version of Hinode? Thank you so much!

**Reply by markdumay** [reply]
Follow the steps of the `npm` tab in [Installation](https://gethinode.com/docs/getting-started/introduction/#installation). Then manually transfer the content and configuration settings to your new repo. The overall setup hasn't changed that much, so it should be relatively easy.

**Reply by horieva** [reply]
Thank you su much. I’m sorry, but I ran into a problem again.

```
 ~/code/mpe/latest/ [main] npm install && npm run mod:update
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: eslint-config-standard@17.1.0
npm ERR! Found: eslint-plugin-n@17.6.0
npm ERR! node_modules/eslint-plugin-n
npm ERR!   dev eslint-plugin-n@"^17.6.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint-plugin-n@"^15.0.0 || ^16.0.0 " from eslint-config-standard@17.1.0
npm ERR! node_modules/eslint-config-standard
npm ERR!   dev eslint-config-standard@"^17.1.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint-plugin-n@16.6.2
npm ERR! node_modules/eslint-plugin-n
npm ERR!   peer eslint-plugin-n@"^15.0.0 || ^16.0.0 " from eslint-config-standard@17.1.0
npm ERR!   node_modules/eslint-config-standard
npm ERR!     dev eslint-config-standard@"^17.1.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/admin/.npm/_logs/2024-05-12T08_29_18_064Z-eresolve-report.txt
```

My Node version:  v19.6.1

**Reply by markdumay** [reply]
Please see https://github.com/gethinode/hinode/issues/901. I'll check if I need to add this to the base repo.

**Comment by horieva** [comment]
And another question: I try to insert images: 

In my `config/_default/config.toml`:
```
[params.slider]
  images = [
    { src = "/img/hero-slide-1.jpeg", alt = "Hero Slide 1" },
    { src = "/img/mpe-logo.svg", alt = "Image 2" }
]
```
In my `layouts/_default/index.html`:
```
    <div class="slider">
        {{ range .Site.Params.slider.images }}
            <div class="slide">
                <img src="{{ .src }}" alt="{{ .alt }}">
            </div>
        {{ end }}
    </div>
```

File `hero-slide-1.jpeg` placed to the `static/img/`. I don’t know why, but `mpe-logo.svg` is displayed but `hero-slide-1.jpeg` no (not found).

**Reply by markdumay** [reply]
You might want to familiarize yourself with Hugo's folder structure. The getting started guide provides a [brief summary of the various folders and their use](https://gethinode.com/guides/editing/#step-2---managing-the-content). Hugo expects [images to be available as either page resource, global resource, or remote resource](https://gohugo.io/content-management/image-processing/). In your example, the image should be present in the `assets/img` folder. Supported image types include `png`, `jpeg`, `gif`,
  `tiff`, `bmp`, and  `webp`.

Strictly speaking, vector graphics `svg` are not considered to be images by Hugo. The regular image processing methods are not available to images of such type. Instead, vector graphics are handled by the browser. You should serve vector graphics files from your `static` folder.


Lastly, I'm not sure what the intent is of your slider, but you might want to take a look at Hinode's [image carousel](https://gethinode.com/docs/components/carousel/). Alternatively, Hinode provides an [image shortcode](https://gethinode.com/docs/components/image/). It prepares so-called image sets, that scale with the viewport and include various optimizations.

**Comment by horieva** [comment]
Thank you very much, I have more questions. :)

~~1. If I want to use the dark/white page color switch, how can I change the logo when switching? For example, mpe-dark.svg & mpe-light.svg?~~ **SOLVED** with added `-light` and `-dark` to the logo file name.

2. In my `params.toml` file, the `breadcrumb = false` parameter, but breadcrumbs still appear on all pages except the homepage. Why?

3. Where is the main page template located? I would like to modify the layout of the homepage, but I can't find it. Sidebar is disabled in the `params.toml` but I see empty sidebar col (md-3) at right on the main page. 

4. Also how can I create a separate template for a new page and where should I specify that I want to use this template when processing that page?

5. About different language content. Am I correct in understanding that if my main site version is in English and not in the /en subdirectory URL, then I should create pages not in `content/en` but in `content`?

Thank you for your detailed responses. Apologies for the newbie questions.

**Reply by markdumay** [reply]
The [breadcrumb setting ](https://gethinode.com/docs/configuration/navigation/#basic-configuration)should be set in the **`navigation`** section of the site parameters (e.g. `config/_default/params.toml`):

```toml
[navigation]
    breadcrumb = true
```

The base layout is defined in `layouts/_default/baseof.thml`. See the [Hugo docs](https://gohugo.io/templates/base/) for more details.

You can define custom page layouts by setting the `type` in the page's front matter. Add the various templates in the `layout/mylayout` folder. See https://gohugo.io/templates/lookup-order/ for more details.

You can decide the content structure that best suits you. If your site supports one language only (e.g. `nl`), than you could omit the subfolder entirely. See https://gohugo.io/content-management/multilingual/ for more details.

**Reply by horieva** [reply]
For some reason, on the main page, the main block is divided into two parts, one of which contains empty space for the Sidebar. I would like to display the main page in the same layout as the other pages of the website. How can I change the layout of the main page of the website?
Thank you

![CleanShot 2024-05-14 at 11 20 09@2x](https://github.com/gethinode/hinode/assets/90963491/3f2b94a5-b1da-4d0e-a8b4-f0d1a55e656e)

**Reply by markdumay** [reply]
The main page (homepage, e.g. `content/_index.md`) of the site is a list page. You can modify its layout in `layouts/_default/list/body.html`. By default, it renders two columns (size `9` and size `3). The next snippet removes the second column entirely:


```go-template
{{- $sections := $.Scratch.Get "sections" -}}
{{- $fullCover := $.Scratch.Get "fullCover" -}}
{{- $breakpoint := $.Scratch.Get "breakpoint" -}}
{{- $layout := .Params.layout -}}

{{ if not .IsHome }}
    {{ with .Title }}<p id="{{ anchorize . }}" class="display-4 mt-5{{ if and $.IsHome site.Params.home.centerHeadline }} text-center{{ end }}">{{ . }}</p>{{ end }}
{{ end }}
{{ if (.Params.menu) }}
    {{- partial "assets/section-menu.html" (dict "page" .) -}}
{{- end -}}
{{- $loading := "" -}}
{{- if or (eq $layout "featured") .IsHome -}}
    {{- if $fullCover }}{{ $loading = site.Params.main.loading }}{{ end }}
{{ end }}
{{- $content := partial "utilities/ProcessContent" (dict "page" . "raw" .RawContent "loading" $loading) -}}
{{- $content | safeHTML -}}
{{ if and (and $sections (eq (len $sections) 1)) (not $content) }}
    <p class="pt-4">{{- T "emptyList" }}.</p>
{{ end }}
```

**Reply by horieva** [reply]
Thank you, you're the best! Latest question: how I can remove blog section on the bottom of the main page?

**Reply by markdumay** [reply]
The sections on the homepage are defined in `home.sections` in `params.toml`. Hinode inserts default values when this entry is empty. Using a dummy value removes all sections from the homepage:

```toml
[home]
    sections = ["dummy"]
```

**Reply by vinser** [reply]
Yes, it removes but writes `{{- T "emptyList" }}`
I had to finalize it with a steel file
```
{{/* Remove home page sections in case if dummy */}}
{{ if .IsHome }}
    {{- if and (eq (len site.Params.home.sections) 1) (eq (index site.Params.home.sections 0) "dummy") -}}
        {{- $sections = first 0 $sections -}}
    {{- end -}}
{{ end }}
```

**Comment by horieva** [comment]
I have another question. 

Is there a way to make the image enlarge upon clicking, like a Magnific Popup, when using the tag `{{< image src="/img/index/image-001.jpg" ratio="1x1" class="rounded img-thumbnail float-end lana-right" >}}`? Thank you.

I like that plugin: https://photoswipe.com/getting-started/ - How can I embed it into the page code correctly?

**Reply by markdumay** [reply]
You will need to add the CSS and JavaScript files of PhotoSwipe to your project. See the pipelines for [styles](https://gethinode.com/docs/advanced-settings/styles/) and [scripts](https://gethinode.com/docs/advanced-settings/scripts/) in the Hinode docs.

You could develop a custom module if you would like to take advantage of automation (e.g. automated dependency updates). I have written a guide on [how to develop a custom module compatible with Hinode](https://gethinode.com/guides/modules/).

**Comment by horieva** [comment]
Today I encountered this problem: when I do a Git push and deploy to Netlify, I get this error on Netlify:

```
5:40:15 AM:             curl -LJO https://github.com/sass/dart-sass/releases/download/1.77.0/dart-sass-1.77.0-linux-x64.tar.gz &&             tar -xf dart-sass-1.77.0-linux-x64.tar.gz &&             rm dart-sass-1.77.0-linux-x64.tar.gz &&             export PATH=/opt/build/repo/dart-sass:$PATH && npm run build
5:40:15 AM:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
5:40:15 AM:                                  Dload  Upload   Total   Spent    Left  Speed
5:40:15 AM:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
5:40:15 AM:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
5:40:16 AM: 100 3674k  100 3674k    0     0  6998k      0 --:--:-- --:--:-- --:--:-- 6998k
5:40:16 AM: curl: Saved to filename 'dart-sass-1.77.0-linux-x64.tar.gz'
5:40:16 AM: > @gethinode/template@0.13.0 prebuild
5:40:16 AM: > npm run clean:public && npm run -s mod:vendor
5:40:16 AM: > @gethinode/template@0.13.0 clean:public
5:40:16 AM: > rimraf public
5:40:17 AM: > @gethinode/template@0.13.0 build
5:40:17 AM: > hugo --gc --minify
5:40:17 AM: Start building sites …
5:40:17 AM: hugo v0.125.7-b1d808bc373f53ad37c8966bb02a6aea095db5f8+extended linux/amd64 BuildDate=2024-05-08T14:46:24Z VendorInfo=gohugoio
5:40:20 AM: ERROR render of "page" failed: "/opt/build/repo/layouts/_default/baseof.html:73:12": execute of template failed: template: _default/single.html:73:12: executing "_default/single.html" at <partialCached "footer/scripts.html" (dict "header" false "page" . "core" true)>: error calling partialCached: "/opt/build/repo/layouts/partials/footer/scripts.html:35:18": execute of template failed: template: partials/footer/scripts.html:35:18: executing "partials/footer/scripts.html" at <$js.Content>: error calling Content: EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
5:40:20 AM: ERROR render of "term" failed: "/opt/build/repo/layouts/_default/baseof.html:73:12": execute of template failed: template: tags/list.html:73:12: executing "tags/list.html" at <partialCached "footer/scripts.html" (dict "header" false "page" . "core" true)>: error calling partialCached: "/opt/build/repo/layouts/partials/footer/scripts.html:35:18": execute of template failed: template: partials/footer/scripts.html:35:18: executing "partials/footer/scripts.html" at <$js.Content>: error calling Content: EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
5:40:20 AM: ERROR render of "404" failed: "/opt/build/repo/layouts/_default/baseof.html:73:12": execute of template failed: template: 404.html:73:12: executing "404.html" at <partialCached "footer/scripts.html" (dict "header" false "page" . "core" true)>: error calling partialCached: "/opt/build/repo/layouts/partials/footer/scripts.html:35:18": execute of template failed: template: partials/footer/scripts.html:35:18: executing "partials/footer/scripts.html" at <$js.Content>: error calling Content: EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
5:40:20 AM: ERROR EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
5:40:20 AM: Total in 2822 ms
5:40:20 AM: Error: error building site: render: failed to render pages: render of "page" failed: "/opt/build/repo/layouts/_default/baseof.html:73:12": execute of template failed: template: _default/single.html:73:12: executing "_default/single.html" at <partialCached "footer/scripts.html" (dict "header" false "page" . "core" true)>: error calling partialCached: "/opt/build/repo/layouts/partials/footer/scripts.html:35:18": execute of template failed: template: partials/footer/scripts.html:35:18: executing "partials/footer/scripts.html" at <$js.Content>: error calling Content: EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
5:40:20 AM: ​
5:40:20 AM: "build.command" failed                                        
5:40:20 AM: ────────────────────────────────────────────────────────────────
5:40:20 AM: ​
5:40:20 AM:   Error message
5:40:20 AM:   Command failed with exit code 1: (https://ntl.fyi/exit-code-1)
5:40:20 AM:               curl -LJO https://github.com/sass/dart-sass/releases/download/1.77.0/dart-sass-1.77.0-linux-x64.tar.gz &&             tar -xf dart-sass-1.77.0-linux-x64.tar.gz &&             rm dart-sass-1.77.0-linux-x64.tar.gz &&             export PATH=/opt/build/repo/dart-sass:$PATH && npm run build
5:40:20 AM: ​
5:40:20 AM:   Error location
5:40:20 AM:   In build.command from a plugin:
5:40:20 AM: ​
5:40:20 AM:               curl -LJO https://github.com/sass/dart-sass/releases/download/1.77.0/dart-sass-1.77.0-linux-x64.tar.gz &&             tar -xf dart-sass-1.77.0-linux-x64.tar.gz &&             rm dart-sass-1.77.0-linux-x64.tar.gz &&             export PATH=/opt/build/repo/dart-sass:$PATH && npm run build
5:40:20 AM: ​
5:40:20 AM:   Resolved config
5:40:20 AM:   build:
5:40:20 AM:     command: |2-
5:40:20 AM: ​
5:40:20 AM:                   curl -LJO https://github.com/sass/dart-sass/releases/download/1.77.0/dart-sass-1.77.0-linux-x64.tar.gz &&             tar -xf dart-sass-1.77.0-linux-x64.tar.gz &&             rm dart-sass-1.77.0-linux-x64.tar.gz &&             export PATH=/opt/build/repo/dart-sass:$PATH && npm run build
5:40:20 AM:     commandOrigin: inline
5:40:20 AM:     environment:
5:40:20 AM:       - DART_SASS_VERSION
5:40:20 AM:       - HUGO_VERSION
5:40:20 AM:       - HUGO_ENV
5:40:20 AM:       - HUGO_ENABLEGITINFO
5:40:20 AM:       - NODE_VERSION
5:40:20 AM:       - NPM_VERSION
5:40:20 AM:     publish: /opt/build/repo/public
5:40:20 AM:     publishOrigin: config
5:40:20 AM:   functionsDirectory: /opt/build/repo/functions
5:40:20 AM:   headers:
5:40:21 AM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
5:40:21 AM:     - for: /*
      values:
        Access-Control-Allow-Origin: '*'
        Content-Security-Policy: |-
          default-src 'self'; child-src 'self' app.netlify.com; script-src 'self' [https://*.netlify.app](https://%2A.netlify.app/) app.netlify.com netlify-cdp-loader.netlify.app [https://*.google-analytics.com](https://%2A.google-analytics.com/) [https://*.googletagmanager.com;](https://%2A.googletagmanager.com;/) style-src 'self' [https://*.netlify.app](https://%2A.netlify.app/) https://fonts.googleapis.com/ https://www.youtube.com;/ object-src 'none'; base-uri 'self'; connect-src 'self'
                          [https://*.google-analytics.com](https://%2A.google-analytics.com/) [https://*.analytics.google.com](https://%2A.analytics.google.com/) [https://*.googletagmanager.com;](https://%2A.googletagmanager.com;/) font-src 'self' [https://*.netlify.app](https://%2A.netlify.app/) https://fonts.gstatic.com;/ frame-src 'self' https://www.youtube-nocookie.com/ https://www.youtube.com/ app.netlify.com; img-src 'self' data: [https://*.netlify.app](https://%2A.netlify.app/) https://i.vimeocdn.com/ https://i.ytimg.com/ [https://*.google-analytics.com](https://%2A.google-analytics.com/) [https://*.googletagmanager.com](https://%2A.googletagmanager.com/) https://tile.openstreetmap.org;/ manifest-src 'self'; media-src 'self'
        Permissions-Policy: geolocation=(), midi=(), sync-xhr=(), microphone=(), camera=(), magnetometer=(), gyroscope=(), fullscreen=(), payment=()
        Referrer-Policy: strict-origin
        Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
        X-Content-Type-Options: nosniff
        X-Frame-Options: SAMEORIGIN
        X-XSS-Protection: 1; mode=block
        cache-control: max-age=0, no-cache, no-store, must-revalidate
  headersOrigin: config
  plugins:
    - inputs: {}
      origin: config
      package: '@gethinode/netlify-plugin-dartsass'
    - inputs: {}
      origin: config
      package: netlify-plugin-hugo-cache-resources
    - inputs: {}
      origin: config
      package: '@netlify/plugin-lighthouse'
5:40:21 AM: Build failed due to a user error: Build script returned non-zero exit code: 2
5:40:21 AM: Failing build: Failed to build site
5:40:21 AM: Finished processing build request in 29.248s
```

**Reply by markdumay** [reply]
The key error seems to be the following line, which indicates there's a build error in your JavaScript pipeline:

```bash
ERROR EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): 
failed to parse Resource "/js/main.bundle.js" as Template:: 
template: /js/main.bundle.js:6835: function "IDLE" not defined
```

Are you absolutely sure the build works locally?

**Reply by horieva** [reply]
Ooops, no I has the error too. But I don't know why that happens. 

Can you look to my repo please? It located in https://github.com/horieva/mpe-bouwtechniek

Local error log is: 

```
ERROR render of "section" failed: "/Users/admin/code/upwork/mpe/git/layouts/_default/baseof.html:73:12": execute of template failed: template: _default/list.html:73:12: executing "_default/list.html" at <partialCached "footer/scripts.html" (dict "header" false "page" . "core" true)>: error calling partialCached: "/Users/admin/code/upwork/mpe/git/layouts/partials/footer/scripts.html:35:18": execute of template failed: template: partials/footer/scripts.html:35:18: executing "partials/footer/scripts.html" at <$js.Content>: error calling Content: EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
ERROR render of "page" failed: "/Users/admin/code/upwork/mpe/git/layouts/_default/baseof.html:73:12": execute of template failed: template: _default/single.html:73:12: executing "_default/single.html" at <partialCached "footer/scripts.html" (dict "header" false "page" . "core" true)>: error calling partialCached: "/Users/admin/code/upwork/mpe/git/layouts/partials/footer/scripts.html:35:18": execute of template failed: template: partials/footer/scripts.html:35:18: executing "partials/footer/scripts.html" at <$js.Content>: error calling Content: EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
ERROR EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
Built in 2188 ms
Error: error building site: render: failed to render pages: render of "page" failed: "/Users/admin/code/upwork/mpe/git/layouts/_default/baseof.html:73:12": execute of template failed: template: _default/single.html:73:12: executing "_default/single.html" at <partialCached "footer/scripts.html" (dict "header" false "page" . "core" true)>: error calling partialCached: "/Users/admin/code/upwork/mpe/git/layouts/partials/footer/scripts.html:35:18": execute of template failed: template: partials/footer/scripts.html:35:18: executing "partials/footer/scripts.html" at <$js.Content>: error calling Content: EXECUTE-AS-TEMPLATE: failed to transform "/js/main.bundle.js" (text/javascript): failed to parse Resource "/js/main.bundle.js" as Template:: template: /js/main.bundle.js:6835: function "IDLE" not defined
```

**Reply by markdumay** [reply]
Your import of PhotoSwipe is incorrect. I submitted a PR that should fix it. Basically, you're trying the ESM version, which is not compatible with the Hugo build system. Instead, you should use the universal module definition (UMD) format. I've also removed most of the unnecessary source files. Lastly, you forgot to import the stylesheet. I've renamed the file to an `.scss` extension and included the import in your `theme.scss` file. I haven't done any additional validation and testing, but at least the build works now.

**Reply by horieva** [reply]
Thank you very much! Hinode has the most wonderful and responsive developer I have ever seen in my life. :)

**Reply by horieva** [reply]
I tried to implement Photoswipe using the simple example located at https://photoswipe.com/getting-started/. 

I added the code "Basic Vanilla JS Example" to `layouts/_default/single/footer.html` and tried to correct the paths in the JS file call. Unfortunately, I still get the error `"Uncaught TypeError: Failed to resolve module specifier "js/photoswipe/photoswipe-lightbox.umd.min.js". Relative references must start with either "/", "./", or "../"."`

At the same time, I temporarily turned off CSRF.

```

<script type="module">
    import PhotoSwipeLightbox from 'js/photoswipe/photoswipe-lightbox.umd.min.js';

const lightbox = new PhotoSwipeLightbox({
  // may select multiple "galleries"
  gallery: '#gallery--getting-started',

  // Elements within gallery (slides)
  children: 'a',

  // setup PhotoSwipe Core dynamic import
  pswpModule: () => import('js/photoswipe/photoswipe.umd.min.js')
});
lightbox.init();
</script>

<div class="pswp-gallery pswp-gallery--single-column" id="gallery--getting-started">
  <a href="https://cdn.photoswipe.com/photoswipe-demo-images/photos/2/img-2500.jpg"
    data-pswp-width="1669"
    data-pswp-height="2500"
    target="_blank">
    <img src="https://cdn.photoswipe.com/photoswipe-demo-images/photos/2/img-200.jpg" alt="" />
  </a>
  <!-- cropped thumbnail: -->
  <a href="https://cdn.photoswipe.com/photoswipe-demo-images/photos/7/img-2500.jpg"
    data-pswp-width="1875"
    data-pswp-height="2500"
    data-cropped="true"
    target="_blank">
    <img src="https://cdn.photoswipe.com/photoswipe-demo-images/photos/7/img-200.jpg" alt="" />
    Cropped
  </a>
  <!-- data-pswp-src with custom URL in href -->
  <a href="https://unsplash.com"
    data-pswp-src="https://cdn.photoswipe.com/photoswipe-demo-images/photos/3/img-2500.jpg"
    data-pswp-width="2500"
    data-pswp-height="1666"
    target="_blank">
    <img src="https://cdn.photoswipe.com/photoswipe-demo-images/photos/3/img-200.jpg" alt="" />
  </a>
  <!-- Without thumbnail: -->
  <a href="http://example.com"
    data-pswp-src="https://cdn.photoswipe.com/photoswipe-demo-images/photos/5/img-2500.jpg"
    data-pswp-width="2500"
    data-pswp-height="1668"
    target="_blank">
    No thumbnail
  </a>
  <!-- wrapped with any element: -->
  <div>
    <a href="https://cdn.photoswipe.com/photoswipe-demo-images/photos/6/img-2500.jpg"
      data-pswp-width="2500"
      data-pswp-height="1667"
      target="_blank">
      <img src="https://cdn.photoswipe.com/photoswipe-demo-images/photos/6/img-200.jpg" alt="" />
    </a>
  </div>
</div>
```

**Reply by markdumay** [reply]
I submitted a PR with a basic working example. The PhotoSwipe example uses dynamic loading. Instead, I used the following basic script `photoswipe.zinit.js`:

```js
const lightbox = new PhotoSwipeLightbox({
  gallery: '#gallery--getting-started',
  children: 'a',
  pswpModule: PhotoSwipe
})

lightbox.init()
```

The above script needs to be loaded **after** the PhotoSwipe core files. I enabled debugging in `params.toml` of the JS files so you can verify the import order of the script files:

```toml
[debugging]
    showJS = true
```

You need to whitelist the  PhotoSwipe styles to avoid them from being purged (`config/postcss.config.js`):

```js
'./assets/scss/theme/photoswipe.scss'
```

Finally, the test file `content/swipe.en.md` contains some unsupported HTML code. To enable it, add the following setting to `config/_default/markup.toml`:

```toml
[goldmark]
    [goldmark.renderer]
        unsafe = true
```

**Comment by horieva** [comment]
How I can add the logo from the `static` to the footer? Like navbar which changes when light/dark is pushed? 

Thank you!

**Reply by markdumay** [reply]
You could check https://github.com/gethinode/docs, it includes a custom footer with such a logo.

**Comment by horieva** [comment]
Trying to create the partial and insert into markdown as described in https://discourse.gohugo.io/t/can-i-call-a-shortcode-from-a-partial/39043/5

Created `layouts/partials/call-us.html`:

```
<div class="hero">
Hello
</div>
```

...and trying to use in in the `services.en.md` like: `{{< call-us >}}`

I get the error: ` failed to extract shortcode: template for shortcode "call-us" not found`

**Reply by markdumay** [reply]
You cannot call a partial directly from within markdown. You will need to create a shortcode instead. The shortcode can optionally call a partial, as described in the linked discussion.

```
# this works
markdown -> shortcode
markdown -> shortcode -> partial

# this does not work
markdown -> partial
```

If you add the following shortcode `layouts/shortcodes/call-us.html` your example should work:

```
partial "call-us.html"
```
